lib/camunda-cloud/features/properties-provider/parts/implementation/FormDefinition.js

```diff
@@ -0,0 +1,212 @@
+import entryFactory from 'bpmn-js-properties-panel/lib/factory/EntryFactory';
+
+import {
+  getBusinessObject,
+  is
+} from 'bpmn-js/lib/util/ModelUtil';
+
+import elementHelper from 'bpmn-js-properties-panel/lib/helper/ElementHelper';
+
+import cmdHelper from 'bpmn-js-properties-panel/lib/helper/CmdHelper';
+
+import {
+  nextId
+} from 'bpmn-js-properties-panel/lib/Utils';
+
+import {
+  createFormKey,
+  getFormDefinition,
+  getUserTaskForm
+} from '../../../../helper/FormsHelper';
+
+const USER_TASK_FORM_PREFIX = 'userTaskForm_';
+
+
+export default function(element, bpmnFactory, translate) {
+  let entries = [];
+
+  function cleanupUserTaskForm(element) {
+
+    const businessObject = getBusinessObject(element),
+          rootElement = getRootElement(element),
+          extensionElements = businessObject.get('extensionElements'),
+          rootExtensionElements = rootElement.get('extensionElements');
+
+    let commands = [];
+
+    // (1) remove form definition
+    const formDefinition = getFormDefinition(element);
+
+    if (!formDefinition) {
+      return commands;
+    }
+
+    commands.push(cmdHelper.removeElementsFromList(
+      element,
+      extensionElements,
+      'values',
+      'extensionElements',
+      [ formDefinition ]
+    ));
+
+    // (2) remove referenced user task form
+    const userTaskForm = getUserTaskForm(element);
+
+    if (!userTaskForm) {
+      return commands;
+    }
+
+    commands.push(cmdHelper.removeElementsFromList(
+      element,
+      rootExtensionElements,
+      'values',
+      'extensionElements',
+      [ userTaskForm ]
+    ));
+
+    return commands;
+  }
+
+  entries.push(entryFactory.textField(translate, {
+    id: 'form-json',
+    label: translate('Form JSON Configuration'),
+    modelProperty: 'formJSON',
+    get: function(element) {
+      const userTaskForm = getUserTaskForm(element);
+
+      return {
+        formJSON: userTaskForm ? userTaskForm.get('body') : ''
+      };
+    },
+    set: function(element, values) {
+
+      const businessObject = getBusinessObject(element),
+            rootElement = getRootElement(element),
+            body = values.formJSON;
+
+      let commands = [],
+          userTaskForm,
+          formId;
+
+      // (0) cleanup user task form + form definition on empty body
+      if (!body || body === '') {
+        return cleanupUserTaskForm(element);
```

Comment:
We don't perform the cleanup when a User Task is morphed into another type or when it's removed. The result is that we end up with redundant extension elements on the process.

Reply:
an alternative could be to handle it via a modeling behavior.

Reply:
That would be a perfect place for this 🏅

Reply:
I'll tackle this